test/CMakeLists.txt

```
@@ -8,30 +8,56 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
-set(DLAF_FORTRAN_MPIRUN "" CACHE STRING "MPI run command.")
-separate_arguments(_DLAF_FORTRAN_MPIRUN UNIX_COMMAND ${DLAF_FORTRAN_MPIRUN})
+# Define test executables
+set(INIT_TEST_EXECUTABLES init)
+set(PXPOTRF_TEST_EXECUTABLES pspotrf pdpotrf pcpotrf pzpotrf)
+set(PXHEEVD_TEST_EXECUTABLES pssyevd pdsyevd pcheevd pzheevd)
+set(PXHEGVX_TEST_EXECUTABLES pssygvx pdsygvx pchegvx pzhegvx)
+
+
+find_program(FYPP_PROGRAM fypp)
+
+if("${FYPP_PROGRAM}" MATCHES ".*-NOTFOUND")
```

I think it should be exactly the same behavior as (unless I'm missing some edge case)

```suggestion
if(NOT FYPP_PROGRAM)
```

https://cmake.org/cmake/help/latest/command/if.html#basic-expressions

---

I tried something similar, but had issues. Mainly because without the `-NOTFOUND` it was going to the `if(<string>` case, and always evaluating to `false`.

---

I'll try your syntax and see if that works (I might have used something slightly different).

---

I think I might have erroneously used `if(NOT ${FYPP_PROGRAM})`, thanks for spotting this!